Tests/Tests/SDWebImageDecoderTests.m

```diff
@@ -120,8 +126,36 @@ - (void)test10ThatAnimatedWebPCoderWorks {
         expect(imageScale).to.equal(scale);
     }];
     
+    expect([[SDWebImageWebPCoder sharedCoder] canEncodeToFormat:SDImageFormatWebP]).to.beTruthy();
     NSData *outputData = [[SDWebImageWebPCoder sharedCoder] encodedDataWithImage:animatedWebPImage format:SDImageFormatWebP];
     expect(outputData).toNot.beNil();
 }
 
+- (void)test20ThatOurGIFCoderWorksNotFLAnimatedImage {
+    NSURL *gifURL = [[NSBundle bundleForClass:[self class]] URLForResource:@"TestImage" withExtension:@"gif"];
+    NSData *gifData = [NSData dataWithContentsOfURL:gifURL];
+    expect([[SDWebImageGIFCoder sharedCoder] canDecodeFromData:gifData]).to.beTruthy();
+    // the IOCoder can also decode GIF
+    expect([[SDWebImageImageIOCoder sharedCoder] canDecodeFromData:gifData]).to.beTruthy();
+    UIImage *gifImage = [[SDWebImageGIFCoder sharedCoder] decodedImageWithData:gifData];
+    expect(gifImage).toNot.beNil();
+    expect(gifImage.images.count).to.beGreaterThan(0);
+    CGSize imageSize = gifImage.size;
+    CGFloat imageScale = gifImage.scale;
+    [gifImage.images enumerateObjectsUsingBlock:^(UIImage * _Nonnull image, NSUInteger idx, BOOL * _Nonnull stop) {
+        CGSize size = image.size;
+        CGFloat scale = image.scale;
+        expect(imageSize.width).to.equal(size.width);
+        expect(imageSize.height).to.equal(size.height);
+        expect(imageScale).to.equal(scale);
+    }];
+    
+    expect([[SDWebImageGIFCoder sharedCoder] decompressedImageWithImage:gifImage data:nil  options:nil]).to.equal(gifImage);
+    
+    expect([[SDWebImageGIFCoder sharedCoder] canEncodeToFormat:SDImageFormatGIF]).to.beTruthy();
+    expect([[SDWebImageImageIOCoder sharedCoder] canEncodeToFormat:SDImageFormatGIF]).to.beTruthy();
+    NSData *outputData = [[SDWebImageGIFCoder sharedCoder] encodedDataWithImage:gifImage format:SDImageFormatGIF];
+    expect(outputData).toNot.beNil();
```

Comment:
Sorry...I forget this task. By the way, what about decode this `outputData` again and check frame count, image size and so on? This can ensure that it works for both decoding & encoding process :D

Reply:
@dreampiggy no prob. I can take care of this. I just copy-pasted from the WebP tests. I will revisit those as well and also fix my comment above (about that `calculateSize` expectancy).

Reply:
Done @dreampiggy.
Per DRY principle, created `verifyCoder:withLocalImageURL:isAnimatedImage:` method that will take a imageURL and a coder and will check decoding and encoding back and forth, sizes and so. Reusing this method for the static WebP image, for the animated WebP image and for the GIF.